marshmallow_enum/__init__.py

```diff
@@ -46,43 +82,41 @@ def __init__(
                 stacklevel=2
             )
 
+        self.enum = enum_type
+        self.by_value = by_value
         self.error = error
 
-        if load_by is None:
-            load_by = LoadDumpOptions.value if by_value else LoadDumpOptions.name
-
-        if not isinstance(load_by, Enum) or load_by not in LoadDumpOptions:
-            raise ValueError(
-                'Invalid selection for load_by must be EnumField.VALUE or EnumField.NAME, got {}'.
-                format(load_by)
-            )
-
-        if dump_by is None:
-            dump_by = LoadDumpOptions.value if by_value else LoadDumpOptions.name
-
-        if not isinstance(dump_by, Enum) or dump_by not in LoadDumpOptions:
-            raise ValueError(
-                'Invalid selection for load_by must be EnumField.VALUE or EnumField.NAME, got {}'.
-                format(dump_by)
-            )
-
-        self.load_by = load_by
-        self.dump_by = dump_by
```

Comment:
It is a backwards breaking change to not set these attributes. Better would be to set them based on the `self.by_value` attr for backwards compatibility.